osbenchmark/worker_coordinator/runner.py

```diff
@@ -1044,8 +1054,11 @@ def calculate_recall(predictions, neighbors, top_k):
             id_field = params.get("id-field-name", "_id")
             candidates = []
             for hit in response_json['hits']['hits']:
-                if id_field in hit:  # Will add to candidates if field value is present
-                    candidates.append(hit[id_field])
+                field_value = _get_field_value(hit, id_field)
+                if field_value:  # Will add to candidates if field value is present
```

Comment:
have we tested this if the field_value is 0 ? will this condition work?

Reply:
Good point. Let me add test case

Reply:
Updated test case and added is None check.